src/org/labkey/test/components/glassLibrary/grids/QueryGrid.java

```diff
@@ -51,10 +58,58 @@ public WebDriver getDriver()
         return _driver;
     }
 
-    public ResponsiveGrid getGrid()
+    // get rowMaps
+
+    public Map<String, String> getRowMap(int rowIndex)
     {
-        _responsiveGrid = new ResponsiveGrid.ResponsiveGridFinder(_driver).find(_queryGridPanel);
-        return _responsiveGrid;
+        return getGrid().getRowMap(rowIndex);
+    }
+
+    public Map<String, String> getRowMap(String containsText)
+    {
+        GridRow row = getGrid().getRow(containsText).orElseThrow(()->
+                new NotFoundException("No row was found with value ["+ containsText +"]"));
+        return row.getRowMap();
+    }
+
+    public Map<String, String> getRowMap(String containsText, String column)
+    {
+        GridRow row = getGrid().getRow(containsText, column).orElseThrow(()->
+                new NotFoundException("No row was found with value ["+ containsText +"] in column ["+ column +"]"));
+        return row.getRowMap();
+    }
+
+    public Map<String, String> getRowMap(Locator.XPathLocator containing)
+    {
+        return getGrid().getRow(containing).orElseThrow(()->
+                new NotFoundException("No row was found with  ["+ containing +"]")).getRowMap();
+    }
+
+    public List<Map<String, String>> getRowMaps()
+    {
+        return getGrid().getRowMaps();
+    }
+
+    // row selection
+
+    public QueryGrid selectRow(int index, boolean checked)
+    {
+        getGrid().getRows().get(index).select(checked);
```

Comment:
Another place where a check could be done that the grid has a select column before trying to click it.